controllers/modelmesh/runtime.go

```diff
@@ -187,15 +187,17 @@ func (m *Deployment) addRuntimeToDeployment(deployment *appsv1.Deployment) error
 
 		// the puller and adapter containers are the same image and are given the
 		// same resources
-		builtInAdapterContainer.Resources = *m.PullerResources
+		if m.PullerResources != nil {
```

Comment:
I don't think this should ever be nil here, might be better to let it fail with deref error since it would imply a bug. Or otherwise at least add an error log in the nil case?

Reply:
The main reason for the change is to support the test changes. Adding use of `BuiltInAdapter` to the tests triggered a nil pointer exception here where the Deployment is not created with all fields being set. I will add a test helper to create a complete `modelmesh.Deployment` for the unit tests (including the test logger I added).

Also I know the controller doesn't support it, but it is valid for a container to not have any resources specified (BestEffort QoS). If it was supported, it would be useful for small dev/test environments